db/compaction/compaction_job.cc

```diff
@@ -1640,6 +1643,7 @@ void CompactionJob::CleanupCompaction() {
         TableCache::Evict(table_cache_.get(), out.meta.fd.GetNumber());
       }
     }
+    sub_compact.io_status.PermitUncheckedError();
```

Comment:
Is there a reason not to do this in the SubCompactionState destructor instead?  If not, please add a comment why this is okay here (because the sub_compact is about to be deleted anyway...)